Edits to Dockerfile to enable paramiko to install

rishiUT opened the pull request:
The edit adds imports to the Dockerfile that allow paramiko to be installed as a dependency without causing build errors.

rishiUT commented:
This should solve issue #871

ivotron commented on src/Dockerfile:
```diff
@@ -3,6 +3,13 @@ FROM python:3.7.7-alpine3.12
+    apk --no-cache add musl-dev && \
+    apk --no-cache add python3-dev && \
+    apk --no-cache add libffi-dev && \
+    apk --no-cache add openssl-dev && \
```
this will make the image size significantly bigger. So one alternative is to remove all the build-time dependencies, but that might complicate things too much. The other option is to install the pre-packaged version of paramiko available in the edge branch of alpine, so something like this should work:

```dockerfile
FROM alpine:20200917

COPY . /popper

RUN apk --no-cache add git py3-pip py3-paramiko
    pip install --no-cache-dir /popper && \
    rm -r /popper
```

rishiUT replied:
Hi @ivotron! We will definitely look further into running with a pre-packaged version of paramiko. I'm not certain how to remove build-time dependencies from a pre-existing library, but I can research that as well if the pre-packaged version doesn't work out. 

Regarding the docstring changes, we will find and remove those as well!

ivotron reviewed (changes requested):
thanks a lot @rishiUT, looking great! please take a look at the comments. Also, seems like some docstring changes were inserted by an IDE (or something alike). Would it be possible to have those in a separate PR? thanks!

ivotron commented:
oh, also, would it be possible to add a section in the docs about how one would run a workflow on a remote host? As a subsection of the "Docker" section would be a good place to write this: https://github.com/getpopper/popper/blob/master/docs/sections/cn_workflows.md#docker

rishiUT commented:
Hello @ivotron! We reverted the changes the IDE made to the docstrings, implemented your suggested fix to the Dockerfile, and re-added the section in the docs regarding running a workflow on a remote host from the commit reverted in issue #871. Is there anything else we can do to repair this issue?

ivotron commented on src/setup.py:
```diff
@@ -22,6 +22,7 @@
         "docker==4.3.1",
         "dockerpty==0.4.1",
         "GitPython==3.1.7",
+        "paramiko==2.7.1",
```
could this be bumped up to 2.7.2?

rishiUT replied:
Done!

ivotron reviewed (commented):
looking great! just left one quick question!

ivotron commented:
sounds good, I've merged it! Thanks a lot for your contribution 🎉
